lib/interferon.rb

```diff
@@ -133,12 +133,55 @@ def read_hosts(sources)
       return hosts
     end
 
-    def update_alerts(destinations, hosts, alerts, groups)
+    def update_alerts(destinations, hosts, alerts, groups, dry_run)
       loader = DestinationsLoader.new([@alerts_repo_path])
       loader.get_all(destinations).each do |dest|
         break if @request_shutdown
         log.info "updating alerts on #{dest.class.name}"
-        update_alerts_on_destination(dest, hosts, alerts, groups)
+        if dry_run
+          dry_run_update_alerts_on_destination(dest, hosts, alerts, groups)
+        else
+          update_alerts_on_destination(dest, hosts, alerts, groups)
+        end
+      end
+    end
+
+    def dry_run_update_alerts_on_destination(dest, hosts, alerts, groups)
```

Comment:
I mean, I don't think this code is unique. You could pass in the logic that you want called and thus have a single if condition.

You could also have `dry_run_update_alerts_on_destination` call `update_alerts_on_destination` and place

```
if !dest.api_errors.empty?
  raise dest.api_errors.to_s
end
```

only in the former.